packages/rancher-logging/generated-changes/patch/values.yaml.patch

```diff
@@ -105,8 +105,12 @@
 +global:
 +  cattle:
 +    systemDefaultRegistry: ""
-+    windows:
-+      enabled: false
++    # Uncomment the below two lines to either enable or disable Windows logging. If this chart is
++    # installed via the Rancher UI, it will set this value to "true" if the cluster is a Windows
++    # cluster. In that scenario, if you would like to disable Windows logging on Windows clusters,
++    # set the value below to "false".
++    # windows:
++    #   enabled: true
```

Comment:
The reason we are doing this is: if a user sets to false then the UI will respect that ? Whereas if it defaults to false then the UI won't ?
Also we need to double check that the UI can indeed set a value that isn't in the values.yaml

Reply:
What happens is a combination of several things:
- The user enters global.cattle.windows.enabled=false
- When you click save, I diff that against the values yaml and save ONLY the things that differ from the chart.  But you've set that false is the default, so what the user entered is dropped since it's the same.
- Then after that I add all the global.* values, and those are set only-if-not-already set.  But it's not set in this case, so I set it to true.

So you commenting out the enabled means that the user's provided `false` is not the default, then it doesn't get dropped, and then it's set so it doesn't get overwritten with `true`.

You can set any arbitrary values in the yaml editor.

Reply:
Explanation:

> spec.chart.values is the values from the chart definition.  spec.values is the user-configured and ui-configured values.  The UI combines the two for the edit screen so you see all the possibilities, then diff's the two and saves only the changes from the chart's definition.
But the chart contains global.cattle.windows.enabled=false, so if a user sets it to false, then it's no different than the default, so it's dropped from the values.  Then at the end the UI adds it's values for global.* if they're not set, which they now aren't because it was dropped.  Which makes the effective value true again.